meeko/molsetup.py

```diff
@@ -4,623 +4,451 @@
-        # element
-        # type
-        # neighbor graph
-        # ignore
-        # update bonds bonds (using the neighbor graph)
-        # If pseudo-atom, update other information, too
-
-
-    # pseudo-atoms
-    def add_pseudo(self, coord=np.array([0.0,0.0,0.0], dtype='float'), charge=0.0,
-            anchor_list=None, atom_type=None, rotatable=False,
-            pdbinfo=None, directional_vectors=None, ignore=False, overwrite=False):
-        """ add a new pseudoatom
-            multiple bonds can be specified in "anchor_list" to support the centroids of aromatic rings
-
-            if rotatable, makes the anchor atom rotatable to allow the pseudoatom movement
+        Parameters
+        ----------
+        atom_index: int
+            atom index to replace with a dummy atom
+
+        Returns
+        -------
+        None
         """
-        idx = self.atom_true_count + len(self.atom_pseudo)
-        if idx in self.coord and not overwrite:
-            print("ADD_PSEUDO> Error: the idx [%d] is already occupied (use 'overwrite' to force)")
-            return False
-        self.atom_pseudo.append(idx)
-        # add the pseudoatom information to the atoms
-        self.add_atom(idx=idx, coord=coord,
-                element=0,
-                charge=charge,
-                atom_type=atom_type,
-                pdbinfo=pdbinfo,
-                ignore=ignore,
-                overwrite=overwrite)
-        # anchor atoms
-        if not anchor_list is None:
-            for anchor in anchor_list:
-                self.add_bond(idx, anchor, order=0, rotatable=rotatable)
-        # directional vectors
-        if not directional_vectors is None:
-            self.add_interaction_vector(idx, directional_vectors)
-        return idx
-
-    def add_bond(self, idx1, idx2, order=0, rotatable=False):
-
-        if not idx2 in self.graph[idx1]:
-            self.graph[idx1].append(idx2)
-        if not idx1 in self.graph[idx2]:
-            self.graph[idx2].append(idx1)
-        self.set_bond(idx1, idx2, order, rotatable)
-
-    def add_rotamer(self, indices_list, angles_list):
-        xyz = self.coord
-        rotamer = {}
-        for (i1, i2, i3, i4), angle in zip(indices_list, angles_list):
-            bond_id = self.get_bond_id(i2, i3)
-            if bond_id in rotamer:
-                raise RuntimeError("repeated bond %d-%d" % bond_id)
-            if not self.bond[bond_id]["rotatable"]:
-                raise RuntimeError("trying to add rotamer for non rotatable bond %d-%d" % bond_id)
-            d0 = calcDihedral(xyz[i1], xyz[i2], xyz[i3], xyz[i4])
-            rotamer[bond_id] = angle - d0
-        self.rotamers.append(rotamer)
-
-    def set_atom_type(self, idx, atom_type):
-        """ set the atom type for atom index
-        atom_type : int or str?
-        return: void
+        blank_atom = Atom(atom_index, is_dummy=True)
+        self.atoms[atom_index] = blank_atom
+        return
+
+    def add_bond(
```

Comment:
For Get_Bond, do we want it to raise an error if the specified bond does not exist, return None with no warnings or errors, or return none with a warning?

Reply:
I think raise error